test/comparisons.jl

```diff
@@ -18,31 +18,39 @@ const INTERVAL_TYPES = [Interval, AnchoredInterval{Ending}, AnchoredInterval{Beg
     # [12]
     #     [45]
     @testset "non-overlapping" begin
-        earlier = convert(A, Interval(1, 2, true, true))
-        later = convert(B, Interval(4, 5, true, true))
-
-        @test earlier != later
-        @test !isequal(earlier, later)
-        @test hash(earlier) != hash(later)
-
-        @test isless(earlier, later)
-        @test !isless(later, earlier)
-
-        @test earlier < later
-        @test !(later < earlier)
-
-        @test earlier ≪ later
-        @test !(later ≪ earlier)
-
-        @test !issubset(earlier, later)
-        @test !issubset(later, earlier)
-
-        @test isempty(intersect(earlier, later))
-        @test_throws ArgumentError merge(earlier, later)
-        @test union([earlier, later]) == [earlier, later]
-        @test !overlaps(earlier, later)
-        @test !contiguous(earlier, later)
-        @test superset([earlier, later]) == Interval(1, 5, true, true)
+        tests = [
+            ((1, 2, true, true), (4, 5, true, true)),
+            ((-∞, 2, true, true), (4, ∞, true, true)),
+        ]
+        for (a, b) in tests
+            @show a
+            @show b
+            earlier = convert(A, Interval(a[1], a[2], a[3], a[4]))
+            later = convert(B, Interval(b[1], b[2], b[3], b[4]))
```

Comment:
These converts seem to fail for AnchoredIntervals, and I currently can't seem to figure out how to fix it. 

Seems to fail here https://github.com/invenia/Intervals.jl/blob/master/src/anchoredinterval.jl#L165 with the error 

```
a = (-∞, 2, true, true)
b = (4, ∞, true, true)
non-overlapping: Error During Test at /Users/fchorney/Documents/repos/julia/Intervals.jl/test/comparisons.jl:20
  Got exception outside of a @test
  TypeError: in Type, in parameter, expected Int64, got InfExtended{Int64}
  Stacktrace:
   [1] convert(::Type{AnchoredInterval{Intervals.Direction{:Right}(),T} where T}, ::Interval{InfExtended{Int64}}) at /Users/fchorney/Documents/repos/julia/Intervals.jl/src/anchoredinterval.jl:167
   [2] top-level scope at /Users/fchorney/Documents/repos/julia/Intervals.jl/test/comparisons.jl:29
   [3] top-level scope at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [4] top-level scope at /Users/fchorney/Documents/repos/julia/Intervals.jl/test/comparisons.jl:21
   [5] top-level scope at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1180
   [6] include at ./boot.jl:328 [inlined]
   [7] include_relative(::Module, ::String) at ./loading.jl:1105
   [8] include(::Module, ::String) at ./Base.jl:31
   [9] include(::String) at ./client.jl:424
   [10] top-level scope at /Users/fchorney/Documents/repos/julia/Intervals.jl/test/runtests.jl:15
   [11] top-level scope at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [12] top-level scope at /Users/fchorney/Documents/repos/julia/Intervals.jl/test/runtests.jl:15
   [13] include at ./boot.jl:328 [inlined]
   [14] include_relative(::Module, ::String) at ./loading.jl:1105
   [15] include(::Module, ::String) at ./Base.jl:31
   [16] include(::String) at ./client.jl:424
   [17] top-level scope at none:6
   [18] eval(::Module, ::Any) at ./boot.jl:330
   [19] exec_options(::Base.JLOptions) at ./client.jl:263
   [20] _start() at ./client.jl:460
```

Reply:
I think this call is occuring `convert(AnchoredInterval{Ending}, Interval(4, ∞, true, true))` which doesn't make sense.

Reply:
I’m unsure why that doesn’t make any sense? Seems like it would work how it used to but now has infinity values in it

Reply:
The reason it's non-sense is the point at which you're setting the one endpoint used in the `AnchoredInterval` is `∞`. Then we're calculating the other endpoint by subtracting `4` from `∞` which is also `∞`. There doesn't seem to be a way to do this conversion without losing information which is why I'm thinking this conversion should error. We should detect this and report a nice error.

Reply:
ah ok I see. Thanks for the explanation!